### ISSUE TITLE
Duplicate output for failing tests; expected behaviour?

### ISSUE THREAD

**Issue by larslockefeer** [facebook/ios-snapshot-test-case#133]
When I ran a snapshot test that failed, the output produced when a Snapshot Test fails was a bit surprising to me. I recorded the reference images on a 64-bit architecture. Then, when I change something to the view and run the tests, I get the following output:

```
Test Case '-[PicnicTests.MessageViewControllerTests testMessageViewControllerWith1Button]' started.
2015-11-11 08:48:27.423 Picnic[30698:20383230] If you have Kaleidoscope installed you can run this command to see an image diff:
ksdiff "/Users/larslockefeer/Projects/Picnic/picnic-ios/PicnicTests/DiffImages/PicnicTests.MessageViewControllerTests/reference_testMessageViewControllerWith1Button@2x.png" "/Users/larslockefeer/Projects/Picnic/picnic-ios/PicnicTests/DiffImages/PicnicTests.MessageViewControllerTests/failed_testMessageViewControllerWith1Button@2x.png"
/Users/larslockefeer/Projects/Picnic/picnic-ios/Pods/FBSnapshotTestCase/FBSnapshotTestCase/SwiftSupport.swift:13: error: -[PicnicTests.MessageViewControllerTests testMessageViewControllerWith1Button] : failed - Snapshot comparison failed: Optional(Error Domain=FBSnapshotTestControllerErrorDomain Code=4 "Images different" UserInfo={NSLocalizedDescription=Images different})
/Users/larslockefeer/Projects/Picnic/picnic-ios/Pods/FBSnapshotTestCase/FBSnapshotTestCase/SwiftSupport.swift:13: error: -[PicnicTests.MessageViewControllerTests testMessageViewControllerWith1Button] : failed - Snapshot comparison failed: Optional(Error Domain=FBSnapshotTestControllerErrorDomain Code=1 "Unable to load reference image." UserInfo={NSLocalizedFailureReason=Reference image not found. You need to run the test in record mode, NSLocalizedDescription=Unable to load reference image., FBReferenceImageFilePathKey=/Users/larslockefeer/Projects/Picnic/picnic-ios/PicnicTests/ReferenceImages_32/PicnicTests.MessageViewControllerTests/testMessageViewControllerWith1Button@2x.png})
Test Case '-[PicnicTests.MessageViewControllerTests testMessageViewControllerWith1Button]' failed (0.214 seconds).
```

Of interest here are lines 4 and 5. Initially, I only noticed the second of these lines and concluded that for some reason, the tests were running on a 32-bit architecture on our CI server or that there was some bug with the suffix generation. It took me a while to see that this is just additional output for the case where `FBSnapshotTestCase` starts looking for a fallback image and fails to find one.

Is this output the expected behaviour for the case where a reference image could be found but failed verification? Or should the second line be omitted?

**Comment by nscoding** [comment]
This is intended, before we completely fail we make an extra test just in case the image from the other architecture can pass the test. There was a discussion on removing this but it will be a breaking change. Most like it will be addressed with the next major version (3.0) and not point update.

Thanks for filling this, I hope it makes sense.

**Comment by samskiter** [comment]
@nscoding this confused me for the last 20 minutes as well.
